### PR TITLE
FIX: useConnectorClient to be enabled only when connected or reconnecting

### PR DESCRIPTION
**Pull request by t0rbik**
This PR aims to prevent `useConnectorClient` running when account is **not** in `connected` state.

Took this along as a quick fix, feel free to close if this is a desired behaviour. Cheers.

### CODE REVIEW

File: packages/react/src/hooks/useConnectorClient.ts [hunk 1]
```diff
@@ -81,7 +81,7 @@ export function useConnectorClient<
     chainId: parameters.chainId ?? chainId,
     connector: parameters.connector ?? connector,
   })
-  const enabled = Boolean(status !== 'disconnected' && (query.enabled ?? true))
+  const enabled = Boolean(status === 'connected' && (query.enabled ?? true))
```
Comment (tmm) [inline packages/react/src/hooks/useConnectorClient.ts:84]: Agree with filtering out `'connecting'`. Let's add `'reconnecting'` here though as it enables optimistic updates.

File: .changeset/short-eyes-hide.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"wagmi": patch
+---
+
+useConnectorClient only enabled when account status is 'connected'.
```
Comment (tmm) [inline .changeset/short-eyes-hide.md:5]:
```suggestion
Updated `useConnectorClient` to enable when status is `'reconnecting'` or `'connected'` (previously was also enabled when status was `'connecting'`).
```

**Review by tmm — CHANGES_REQUESTED** [review]
Thanks! Let's add the same change (and changeset) for the [`useConnectorClient` Vue Composable](https://github.com/wevm/wagmi/blob/main/packages/vue/src/composables/useConnectorClient.ts#L100) too.

**Comment by t0rbik** [comment]
Thank you, I've applied the suggestions.

**Pull request merged**